src/sgame/sg_entities.cpp

```diff
@@ -36,10 +36,78 @@ Maryland 20850 USA.
 #include "sg_local.h"
 #include "sg_entities.h"
 #include "CBSE.h"
+#include "sgame/lua/Interpreter.h"
 
 #include <glm/geometric.hpp>
 #include <glm/gtx/norm.hpp>
 
+
+static std::vector<std::string> eventNames = { // in sync with `enum gentityCallEvent_t`
+	"default",
+	"custom",
+	"free",
+	"call",
+	"act",
+	"use",
+	"die",
+	"reach",
+	"reset",
+	"touch",
+	"enable",
+	"disable"
+};
+
+static const Str::StringRef EventToString( gentityCallEvent_t eventType )
+{
+	size_t index = eventType;
+	return eventNames[ index ];
+}
+
+static void CallLuaEntityHandler( gentity_t *self, Str::StringRef eventName )
+{
+	lua_State *L = Lua::State();
+	// the lua state might not be ready yet
+	if ( L != nullptr )
+	{
+		lua_getglobal( L, "EntityHandlers" );
+		if ( lua_istable( L, -1 ) )
+		{
+			int entityNum = self->num();
+			if ( lua_geti( L, -1, entityNum ) == LUA_TFUNCTION )
+			{
+				lua_pushstring( L, eventName.c_str() );
+				Log::Notice( "executing lua handler for entity %d", entityNum );
```

Comment:
Seems too noisy, should be removed or decreased in log level

Reply:
Using `Log::Verbose` now.